pkg/apis/monitoring/v1alpha1/prometheus_operator_types.go

```diff
@@ -26,3 +27,37 @@ type PrometheusOperatorSpec struct {
 	// Image of config-reloader sidecar container
 	ConfigmapReloaderImage string `json:"configmapReloaderImage,omitempty"`
 }
+
+type PrometheusProvisioning struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PrometheusProvisioningSpec `json:"spec"`
+}
+
+// PrometheusProvisioningSpec extends PrometheusSpec from prometheus-operator by adding fields necessary for deployment
+type PrometheusProvisioningSpec struct {
+	prom.PrometheusSpec `json:",inline"`
+	// Namespace to deploy prometheus statefulset in.
+	Namespace string `json:"namespace,omitempty"`
+	// Namespaces to allowed to be watched by prometheus
+	// +optional
+	// kubebuilder:default=["default","kube-system"] // TODO: how can we include a field value that needs to be defined by user? // FIXME(paulfantom): controller-gen doesn't like defaulting arrays
+	Namespaces []string `json:"namespaces,omitempty"`
```

Comment:
What if we expose this in top-level KubePrometheus CRD and somehow (I don't know yet how) use it to configure prometheus RBAC? This way we won't need to extend prometheus CRD and we could use prometheus-operator native CRD.

Reply:
Makes sense!

Reply:
argh, RBAC is specific to prometheus instance and it cannot be global :(

I wonder if we could hide this and configure namespaces based on `serviceMonitorSelector`, `podMonitorSelector`, `probeNamespaceSelector`, and `ruleNamespaceSelector` from Prometheus CR. This way we don't really need any other field and UX should be better :thinking:

Reply:
Yeah personally I just always configure all to discover everything :man_shrugging:

Reply:
Not sure if I get it, but what is the problem of keeping this here?

I'd only change the field name to make it clearer
```suggestion
	MonitoredNamespaces []string `json:"MonitoredNamespaces,omitempty"`
```

Reply:
The main issue is that we wanted to use prometheus-operator CRDs and create as few new CRDs as possible. Adding new fields means we would need to create new CRD.